### PR TITLE
[Keyboard] Lime by HellTM

### PR DESCRIPTION
**Pull request by HellSingCoder**
a 6×4+6 keys column-staggered split keyboard with encoder and joystick support
https://github.com/HellSingCoder/LimeKeyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lime/lime.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline keyboards/lime/lime.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/lime/readme.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    qmk compile -kb lime -km default
+    qmk flash -kb lime -km default
+
+Flash the default keymap: 
+
+    make lime:default:avrdude
+
+Press reset button on he keyboard when asked.
+
+Disconnect the first half, connect the second one and repeat the process.
```
Comment (drashna) [inline keyboards/lime/readme.md:28]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/lime/rev1/config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#pragma once
```
Comment (drashna) [inline keyboards/lime/rev1/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/lime/rev1/config.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x0666
```
Comment (drashna) [inline keyboards/lime/rev1/config.h:4]: I am genuinely surprised that this isn't taken already.  Though, the low number worries me.  It doesn't need to be changed, though.

File: keyboards/lime/rev1/config.h [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+#   define JOYSTICK_AXES_COUNT 2
+#   define JOYSTICK_BUTTON_COUNT 1
+#   define JOYSTICK_AXES_RESOLUTION 8
+#endif
+
+#define TAP_CODE_DELAY 10
+
+/* communication between sides */
+#define USE_SERIAL
+#define SERIAL_USE_MULTI_TRANSACTION
```
Comment (drashna) [inline keyboards/lime/rev1/config.h:43]: Deprecated, is the default behavior.
```suggestion
```

File: keyboards/lime/rev1/rev1.c [hunk 6]
```diff
@@ -0,0 +1 @@
+#include "lime.h"
```
Comment (drashna) [inline keyboards/lime/rev1/rev1.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/lime/rev1/rev1.h [hunk 7]
```diff
@@ -0,0 +1,25 @@
+#pragma once
```
Comment (drashna) [inline keyboards/lime/rev1/rev1.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/lime/rev1/rules.mk [hunk 8]
```diff
@@ -0,0 +1,3 @@
+ENCODER_ENABLE = yes
+ENCODER_RIGHT_ENABLE = yes
+OLED_DRIVER_ENABLE = yes
```
Comment (drashna) [inline keyboards/lime/rev1/rules.mk:3]:
```suggestion
OLED_ENABLE = yes
OLED_DRIVER = SSD1306
```

File: keyboards/lime/rev1/config.h (hunk 4 above)
Comment (HellSingCoder) [reply keyboards/lime/rev1/config.h:20]: To be honest, me too. I'll change this in the future.

**Comment by HellSingCoder** [comment]
@drashna I moved the encoder code in the c file as that was the only one that didn't cause any issues

**Comment by HellSingCoder** [comment]
Any other change requests before I leave for he holidays?

**Comment by HellSingCoder** [comment]
@zvecr @fauxpark Can you please review this?

File: keyboards/lime/readme.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    make lime:default
+    
+        or
+        
+    qmk compile -kb lime -km default
+    qmk flash -kb lime -km default
+
+Flash the default keymap: 
+
+    make lime:default:avrdude
```
Comment (fauxpark) [inline keyboards/lime/readme.md:26]:
```suggestion
    make lime:default

Flashing example for this keyboard:

    make lime:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**